examples/hal/quad/main.rs

```diff
@@ -105,40 +120,31 @@ fn main() {
     let surface_format = surface
         .capabilities_and_formats(&adapter.physical_device)
         .1
-        .map_or(
-            f::Format::Rgba8Srgb,
-            |formats| {
-                formats
-                    .into_iter()
-                    .find(|format| {
-                        format.base_format().1 == ChannelType::Srgb
-                    })
-                    .unwrap()
-            }
-        );
+        .map_or(f::Format::Rgba8Srgb, |formats| {
+            formats
+                .into_iter()
+                .find(|format| format.base_format().1 == ChannelType::Srgb)
+                .unwrap()
+        });
+
+    let memory_types = adapter.physical_device.memory_properties().memory_types;
+    let limits = adapter.physical_device.limits();
 
-    let memory_types = adapter
-        .physical_device
-        .memory_properties()
-        .memory_types;
-    let limits = adapter
-        .physical_device
-        .limits();
+    let physical_device = adapter.physical_device.clone();
 
     // Build a new device and associated command queues
-    let (device, mut queue_group) =
-        adapter.open_with::<_, hal::Graphics>(1, |family| {
-            surface.supports_queue_family(family)
-        }).unwrap();
+    let (mut device, mut queue_group) = adapter
+        .open_with::<_, hal::Graphics>(1, |family| surface.supports_queue_family(family))
+        .unwrap();
 
-    let mut command_pool = device.create_command_pool_typed(&queue_group, pool::CommandPoolCreateFlags::empty(), 16);
-    let mut queue = &mut queue_group.queues[0];
+    let mut command_pool =
+        device.create_command_pool_typed(&queue_group, pool::CommandPoolCreateFlags::empty(), 16);
 
-    println!("Surface format: {:?}", surface_format);
-    let swap_config = SwapchainConfig::new()
-        .with_color(surface_format)
-        .with_image_usage(i::Usage::COLOR_ATTACHMENT);
-    let (mut swap_chain, backbuffer) = device.create_swapchain(&mut surface, swap_config);
+    let (width, height) =
+        surface.update_dimensions(pixel_width as _, pixel_height as _, &physical_device);
```

Comment:
oh, what happens here? why do we get different width/height? why do we need to update the dimensions, even?